components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/ApiResourceManager.java

```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * Add API resource.
+     *
+     * @param apiResource  API resource.
+     * @param tenantDomain Tenant domain.
+     * @return
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    ApiResource addApiResource(ApiResource apiResource, String tenantDomain) throws ApiResourceManagementException;
+
+    /**
+     * Delete API resource by id.
+     *
+     * @param apiResourceId API resource id.
+     * @param tenantDomain  Tenant domain.
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    void deleteApiResourceById(String apiResourceId, String tenantDomain) throws ApiResourceManagementException;
+
+    /**
+     * Update API resource.
+     *
+     * @param apiResource  API resource.
+     * @param tenantDomain Tenant domain.
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    void updateApiResource(ApiResource apiResource, List<Scope> addedScopes, List<String> removedScopes,
+                           String tenantDomain) throws ApiResourceManagementException;
+
+    /**
+     * Get API resource by identifier.
+     *
+     * @param apiResourceIdentifier API resource identifier.
+     * @param tenantDomain          Tenant domain.
+     * @return An <code>ApiResource</code>.
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    ApiResource getApiResourceByIdentifier(String apiResourceIdentifier, String tenantDomain)
+            throws ApiResourceManagementException;
+
+    /**
+     * Get Api scopes by id.
+     *
+     * @param apiResourceId Api resource id.
+     * @param tenantDomain  Tenant domain.
+     * @return List of <Code>Scope</Code>.
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    List<Scope> getApiScopesById(String apiResourceId, String tenantDomain) throws ApiResourceManagementException;
+
+    /**
+     * Delete API scopes by id.
+     *
+     * @param apiResourceId Api resource id.
+     * @param tenantDomain  Tenant domain.
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    void deleteApiScopesById(String apiResourceId, String tenantDomain) throws ApiResourceManagementException;
+
+    /**
+     * Delete API scope by scope id.
+     *
+     * @param apiResourceId Api resource id.
+     * @param scopeId       Scope id.
+     * @param tenantDomain  Tenant domain.
+     * @throws ApiResourceManagementException API resource management exception.
+     */
+    void deleteApiScopeByScopeId(String apiResourceId, String scopeId, String tenantDomain)
+            throws ApiResourceManagementException;
+
+    /**
+     * Search API Resource in the tenant domain.
+     *
+     * @param apiResourceIdList API Resource Id list.
+     * @param attributes        Attributes to be returned.
+     * @param tenantDomain      Tenant domain.
```

Comment:
Lets add @return . Check for similar places

Reply:
https://github.com/wso2/carbon-identity-framework/pull/4854/commits/8dd16447d6000412e3ff2dd1b2a9ed78249f2d13